### PR TITLE
cpdp-1466: fixed  back links

### PR DESCRIPTION
**Pull request by AndyHWChung**

### CODE REVIEW

File: app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/AddressController.scala [hunk 1]
```diff
@@ -77,7 +77,8 @@ trait AddressController[A <: AddressJourneyType] {
   protected val selectAddressSubmitCall: Call
   protected val continueCall: Call
 
-  protected val enterPostcodePageBackLink: A => Call = _ => isUkCall
+  protected lazy val enterUkAddressBackLinkCall: A => Call = backLinkCall
+  protected val enterPostcodePageBackLink: A => Call       = _ => isUkCall
```
Comment (mikail-khan) [inline app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/AddressController.scala:81]: why is one lazy and the other isn't?

Comment (AndyHWChung) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/AddressController.scala:81]: I made the new one lazy cos I wanted it lazy in the controller I'm overwriting it in - I can take a look at the non lazy one to see what's going on there

Comment (AndyHWChung) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/AddressController.scala:81]: though maybe the one I added in doesn't need to be lazy after all...

Comment (AndyHWChung) [reply app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/AddressController.scala]: turns out it didn't need to be lazy - tested ok with it being non-lazy

**Pull request merged**